scripts/upgrade.ts

```diff
@@ -1,54 +1,228 @@
+  }
+
+  const initStateContract = await migrationSteps.getInitContract({
+    contractNameOrAbi: require("../helpers/StateV2_0_abi_2_1.json"),
+    address: deployInfo.state,
+  });
+
+  const result1 = await getDataFromContract(
+    initStateContract,
+    testId,
+    testState,
+    `data-before-upgrade.${network}.json`
+  );
+
+  // const statesWithProofs = [
+  //   require("../test/state/data/user_state_genesis_transition.json"),
+  //   require("../test/validators/common-data/issuer_genesis_state.json"),
+  //   require("../test/validators/common-data/issuer_next_state_transition.json"),
+  //   require("../test/state/data/user_state_next_transition.json"),
+  // ];
+  // await migrationSteps.populateData(initStateContract, statesWithProofs);
+
+  console.log("============= Start: upgradeToStateV2_migration =============");
+  const { state: stateMigration, verifier } = await stateDeployHelper.upgradeToStateV2_migration(
+    initStateContract.address
+  );
+  console.log("============= Finish: upgradeToStateV2_migration =============");
+
+  console.log("============= Start: initForMigration =============");
+  await stateMigration.initForMigration(verifier.address);
+  console.log("============= Finish: initForMigration =============");
+
+  const logHistory = await migrationSteps.readEventLogData(
+    initStateContract,
+    startBlock,
+    1000,
+    "StateUpdated",
+    `${network}.eventLog.json`
+  );
+  const genesisStatePublishLog = {};
+
+  await migrationSteps.migrateData(
+    logHistory,
+    async (args) => {
+      const stateInfos = await stateMigration.getStateInfoHistoryById(args.id);
+      const receipts: unknown[] = [];
+      if (!genesisStatePublishLog[args.id]) {
+        const genesisState = stateInfos.find(
+          (info) =>
+            info.createdAtTimestamp.toString() === "0" && info.createdAtBlock.toString() === "0"
+        );
+        if (!genesisState) {
+          throw new Error("Genesis state not found");
+        }
+        genesisStatePublishLog[genesisState.id] = true;
+        const tx = await stateMigration.addStateWithTimestampAndBlock(
+          genesisState.id,
+          genesisState.state,
+          genesisState.createdAtTimestamp,
+          genesisState.createdAtBlock
+        );
+        const receipt = await tx.wait();
+        console.log("published genesis state tx with hash: ", receipt.transactionHash);
+        receipts.push(receipt);
+      }
+
+      const tx = await stateMigration.addStateWithTimestampAndBlock(
+        args.id,
+        args.state,
+        args.timestamp,
+        args.blockN
+      );
+      const receipt = await tx.wait();
+      receipts.push(receipt);
+
+      console.log("published state tx with hash: ", receipt.transactionHash);
+
+      return receipts;
+    },
+    `migration.results.${network}.json`
+  );
+
+  const { state: stateV2 } = await migrationSteps.upgradeContract(initStateContract);
+
+  const result2 = await getDataFromContract(
```

Comment:
Why `getDataFromContract` is not a part of migrationSteps instance but `upgradeContract` is?
The same question for `checkData` method.

Additional question:
`checkData` and `getDataFromContract` looks to be more general use case than even data migration, which is definitely not the case for every contract upgrade. So, should it somehow be declared/implemented at the abstract contract level, the interface level?